### ISSUE TITLE
Encrypted Secrets in secret manager should be rotated atleast annually and encrypted with a KMS key

### ISSUE THREAD

**Issue by manjulaK** [awslabs/aws-dataall#443]
### Describe the bug

Encrypted Secrets in secret manager should be rotated atleast annually and encrypted with a KMS key. If the rotation is not an option then consider storing the secret under systems manager parameter store. 

### How to Reproduce

```
*P.S. Please do not attach files as it's considered a security risk. Add code snippets directly in the message body as much as possible.*
```

### Expected behavior

Encrypted Secrets in secret manager should be rotated atleast annually and encrypted with a KMS key. If the rotation is not an option then consider storing the secret under systems manager parameter store.

### OS

All

### Python version

3.1

### AWS data.all version

v1.3,v1.4,v1.5

### Additional context

As per security best practice for secrets management the above need to be met.

**Comment by dlpzx** [comment]
Hi @manjulaK,

I have been looking at this issue, I think you are referring to `dataall-pivot-role-name-<envname>` and `dataall-externalId-<envname>`. They are both encrypted with a different KMS key see `deploy/stacks/secrets_stack.py` 

They are constant values that are referenced in the code and in the environments stack, that's why we do not rotate them.

For the case of `dataall-pivot-role-name-<envname>`, it is more a parameter than a secret, so I will take your suggestion and replace it by an SSM parameter

For `dataall-externalId-<envname>` I am considering different options as it takes advantage of the aws secrets manager `SecretStringGenerator` to get created.

**Comment by noah-paige** [comment]
Closing issue - implemented in v1.6
